lib/js/components/ProgressBar/ProgressBar.vue

```diff
@@ -239,55 +247,79 @@ $progress-bar-layers: (
 		margin-left: $space-xxxxs;
 	}
 
-	&__result {
+	&__range {
+		@each $class, $color-name in $progress-bar-range-colors {
+			&.-#{$class} {
+				background: $color-name;
+			}
+		}
+
 		height: 100%;
 		position: absolute;
 		top: 0;
 	}
+
+	&__badge {
+		@each $class, $colors-map in $progress-bar-badge-colors {
+			&.-#{$class} {
+				background: map-get($colors-map, 'background');
+				color: map-get($colors-map, 'icon');
+			}
+		}
+
+		align-items: center;
+		border-radius: 50%;
+		border: 1px solid $color-inverted-border;
+		display: flex;
+		height: $progress-bar-badge-size;
+		justify-content: center;
+		margin-left: -$progress-bar-badge-size / 2;
+		margin-top: -$progress-bar-badge-size / 2;
+		position: absolute;
+		top: 50%;
+		width: $progress-bar-badge-size;
+
+		&.-small {
+			height: $progress-bar-badge-size-small;
+			margin-left: -$progress-bar-badge-size-small / 2;
+			margin-top: -$progress-bar-badge-size-small / 2;
+			width: $progress-bar-badge-size-small;
+		}
+	}
 }
 </style>
 
 <script lang="ts">
 import { PropType } from 'vue';
 import {
-	PROGRESS_BAR_COLORS,
 	PROGRESS_BAR_SIZES,
 	PROGRESS_BAR_RADII,
 	PROGRESS_BAR_LAYOUTS,
-	PROGRESS_BAR_COLOR_SCHEMES,
-	PROGRESS_BAR_LAYERS,
 	ProgressBarRange,
+	PROGRESS_BAR_LABEL_TEXT_SIZES,
+	PROGRESS_BAR_BADGE_COLORS,
 } from './ProgressBar.consts';
 
+import DsIcon, { ICONS, ICON_SIZES } from '../Icons/Icon';
+
 export default {
 	name: 'ProgressBar',
+	components: {
+		DsIcon,
+	},
 	props: {
-		numberOfLayers: {
-			type: Number,
-			default: PROGRESS_BAR_LAYERS.ONE,
-			validator(size) {
-				return Object.values(PROGRESS_BAR_LAYERS).includes(size);
-			},
-		},
-		colorScheme: {
+		size: {
 			type: String,
-			default: PROGRESS_BAR_COLOR_SCHEMES.DEFAULT,
+			default: PROGRESS_BAR_SIZES.MEDIUM,
 			validator(size) {
-				return Object.values(PROGRESS_BAR_COLOR_SCHEMES).includes(size);
-			},
-		},
-		color: {
-			type: String,
-			default: PROGRESS_BAR_COLORS.INFO,
-			validator(color) {
-				return Object.values(PROGRESS_BAR_COLORS).includes(color);
+				return Object.values(PROGRESS_BAR_SIZES).includes(size);
 			},
 		},
-		size: {
+		labelTextSize: {
 			type: String,
-			default: PROGRESS_BAR_SIZES.MEDIUM,
+			default: PROGRESS_BAR_LABEL_TEXT_SIZES.SMALL,
 			validator(size) {
-				return Object.values(PROGRESS_BAR_SIZES).includes(size);
+				return Object.values(PROGRESS_BAR_LABEL_TEXT_SIZES).includes(size);
 			},
 		},
 		ranges: {
```

Comment:
What do you think about adding a validator here? Maybe we should add some range limits, so https://github.com/bethinkpl/design-system/blob/d7b039c8615d047c040658a80faa8dfc263d915f/lib/js/components/ProgressBar/ProgressBar.vue#L44 doesn't blow up?

Reply:
[`01deed9` (#268)](https://github.com/bethinkpl/design-system/pull/268/commits/01deed93b0ecf013ea69b1a6db761ae6a39babe4)